kernel_tuner/core.py

```diff
@@ -346,6 +348,12 @@ def benchmark_default(self, func, gpu_args, threads, grid, result):
             obs for obs in self.dev.observers if not isinstance(obs, ContinuousObserver)
         ]
 
+        for obs in self.prologue_observers:
```

Comment:
This will call `run_kernel` for each prologue observer separately. Would it be possible to only call `run_kernel` once and call `before_start` and `after_finish` for all prologue observers? Or would that conflict somehow?

Reply:
There are some cases where this may be useful, for instances in combination with the `AccuracyObserver`, which doesn't care about performance and can therefore safely 'observe' the same kernel run as the `NCUObserver`. We would need a mechanism to keep track if an observer prefers/requires to be run in isolation, which complicates things. Call `run_kernel` separate for each `PrologueObserver` circumvents this problem altogether (at the cost of potentially some unneeded redundant `run_kernel` calls).

Reply:
There is some logic in `nvmetrics` that causes some metrics to be evaluated in isolation (requiring multiple profiling passes). We could implement something similar, like a flag that indicates whether a `PrologueObserver` needs to run in isolation. The code in `core.py` should schedule accordingly.

I just feel that it may be a bit overkill for the current PR, as we don't have any other `PrologueObserver`'s yet.

Reply:
Agree. The idea of an isolation flag is useful. Maybe when we have more `PrologueObserver`s we can revisit this issue, but for now we keep it like this.